testdata/testoutput2

```diff
@@ -19612,6 +19612,693 @@ Failed: error -49 at offset 3 in replacement: unknown substring
+          op: ^
+        ]
+        Ket
+        Ket
+        End
+------------------------------------------------------------------
+    b
+ 0: b
+\= Expect no match
+    A
+No match
+    a
+No match
+    c
+No match
+
+/[  ^  a[  ^  b]  ]/B,alt_extended_class
+------------------------------------------------------------------
+        Bra
+        eclass[
+          cls:[ ^a]
+          cls:[ ^b]
+          op: ||
+          cls:[ ]
+          op: ||
+        ]
+        Ket
+        End
+------------------------------------------------------------------
+    \x20
+ 0:  
+    ^
+ 0: ^
+    a
+ 0: a
+    b
+ 0: b
+\= Expect no match
+    c
+No match
+
+/[a-c--b]+/B,alt_extended_class
+------------------------------------------------------------------
+        Bra
+        eclass[
+          cls:[a-c]
+          cls:[b]
+          op: --
+        ]++
+        Ket
+        End
+------------------------------------------------------------------
+    ac
+ 0: ac
+    a
+ 0: a
+\= Expect no match
+    b
+No match
+
+/[a-c--b]{2,3}/B,alt_extended_class
+------------------------------------------------------------------
+        Bra
+        eclass[
+          cls:[a-c]
+          cls:[b]
+          op: --
+        ]{2,3}+
+        Ket
+        End
+------------------------------------------------------------------
+    ac
+ 0: ac
+    cac
+ 0: cac
+\= Expect no match
+    a
+No match
+    bb
+No match
+
+/x[a-c--b]+y/B,alt_extended_class
+------------------------------------------------------------------
+        Bra
+        x
+        eclass[
+          cls:[a-c]
+          cls:[b]
+          op: --
+        ]++
+        y
+        Ket
+        End
+------------------------------------------------------------------
+    xacy
+No match
+    xaay
+No match
+    xay
+No match
+\= Expect no match
+    zacy
+No match
+    xacz
+No match
+    xy
+No match
+    xby
+No match
+
+/[a--/alt_extended_class
+Failed: error 106 at offset 4: missing terminating ] for character class
+
+/[a--a/alt_extended_class
+Failed: error 106 at offset 5: missing terminating ] for character class
+
+/[a--[a/alt_extended_class
+Failed: error 106 at offset 6: missing terminating ] for character class
+
+/[a--[a]/alt_extended_class
+Failed: error 212 at offset 7: missing terminating ] for character class (note '[' must be escaped under PCRE2_ALT_EXTENDED_CLASS)
+
+/[a--[a]--/alt_extended_class
+Failed: error 212 at offset 9: missing terminating ] for character class (note '[' must be escaped under PCRE2_ALT_EXTENDED_CLASS)
+
+/[a--]/alt_extended_class
+Failed: error 210 at offset 5: expected operand after operator in character class
+
+/[--a]/alt_extended_class
+Failed: error 209 at offset 3: unexpected operator in character class (no preceding operand)
+
+/[--]/alt_extended_class
+Failed: error 209 at offset 3: unexpected operator in character class (no preceding operand)
+
+/[a---b]/alt_extended_class
+Failed: error 208 at offset 5: invalid operator in character class
```

Comment:
Should be `[a--[-b]]`

Reply:
Why? We're following UTS#18 syntax here, not inventing our own. Other implementations forbid tripled operator characters.

A single `-` hyphen is either a literal or a range separator, with complex disambiguation. A double hyphen `--` is always an operator (never one end of a range: `[--a]` used to mean "characters from a to hyphen" but in UTS#18 syntax it's invalid and you must instead use `[\--a]`). A triple hyphen is just invalid.

Reply:
There is no tripled operator here. `[#--]` and `[--a]` are currently valid.

It looks like unicode recommendation lists several chars, which cannot be in a pattern without `\`. Do you have tests for them?